samples/bot-tag-mention/csharp/Dialogs/MainDialog.cs

```diff
@@ -0,0 +1,143 @@
+﻿// <copyright file="MainDialog.cs" company="Microsoft">
+// Copyright (c) Microsoft. All rights reserved.
+// </copyright>
+
+using System;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+using AdaptiveCards.Templating;
+using Microsoft.Bot.Builder;
+using Microsoft.Bot.Builder.Dialogs;
+using Microsoft.Bot.Builder.Teams;
+using Microsoft.Bot.Schema;
+using Microsoft.Bot.Schema.Teams;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.Logging;
+using Newtonsoft.Json;
+
+namespace Microsoft.BotBuilderSamples
+{
+    public class MainDialog : LogoutDialog
+    {
+        protected readonly ILogger _logger;
+
+        public MainDialog(IConfiguration configuration, ILogger<MainDialog> logger)
+            : base(nameof(MainDialog), configuration["ConnectionName"])
+        {
+            _logger = logger;
+
+            AddDialog(new OAuthPrompt(
+                nameof(OAuthPrompt),
```

Comment:
Not sure if we are using auth related stuff in this sample.

Reply:
Yes, we are using for graph client.